frontend/src/main/web/lib/stores/GlossaryStore.js

```diff
@@ -0,0 +1,522 @@
+import Dispatcher from '../dispatchers/GlossaryDispatcher';
+import assign from 'object-assign';
+import {EventEmitter} from 'events';
+import {Promise} from 'es6-promise';
+import Request from 'superagent';
+import Configs from '../constants/Configs';
+import {GlossaryActionTypes} from '../constants/ActionTypes';
+import GlossaryHelper from '../utils/GlossaryHelper'
+import StringUtils from '../utils/StringUtils'
+import DateHelpers from '../utils/DateHelper'
+import _ from 'lodash';
+
+var SIZE_PER_PAGE = 5000, CHANGE_EVENT = "change",
+    MAX_LISTENER = 100; //number of listener for GlossaryStore (default is 11)
+
+EventEmitter.prototype.setMaxListeners(MAX_LISTENER);
+
+var _state = {
+  canAddNewEntry: false,
+  canUpdateEntry: false,
+  localeOptions: [],
+  srcLocale: null,
+  selectedTransLocale: null,
+  locales: null,
+  glossary: {},
+  original_glossary: {},
+  glossaryResId: [],
+  page:1,
+  filter: '',
+  focusedRow: {
+    rowIndex: -1
+  },
+  uploadFile: {
+    show: false,
+    status : -1,
+    file: null,
+    transLocale: null
+  },
+  sort: {
+    src_content: true
+  },
+  totalCount: 0,
+  newEntry: {
+    term: '',
+    pos: '',
+    description: '',
+    isSaving: false,
+    show: false,
+    isAllowSave: false
+  }
+};
+
+function localesStatAPIUrl() {
+  return Configs.baseUrl + "/glossary/info" + Configs.urlPostfix
+}
+
+function loadLocalesStats() {
+  var url = localesStatAPIUrl();
+
+  return new Promise(function(resolve, reject) {
+    Request.get(url)
+      .set("Cache-Control", "no-cache, no-store, must-revalidate")
+      .set('Accept', 'application/json')
+      .set("Pragma", "no-cache")
+      .set("Expires", 0)
+      .end(function (err, res) {
+        if(err != null && err.error === true) {
+          console.error(url, err);
+        }
+        if(res != null) {
+          if (res.error) {
+            console.error(url, res.status, res.error.toString());
+            reject(Error(res.error.toString()));
+          } else {
+            resolve(res['body']);
+          }
+        }
+        resolve(null);
+      })
+  });
+}
+
+function processLocalesStatistic(serverResponse) {
+  _state['locales'] = {};
+  _state['localeOptions'] = [];
+  _state['srcLocale'] = serverResponse['srcLocale'];
+  resetCache();
+
+  _.forEach(serverResponse['transLocale'], function(transLocale) {
+    _state['locales'][transLocale.locale.localeId] = transLocale;
```

Comment:
A note about access with `.` vs. access with `[ ]`: the only time to use the latter is cases like this where the key you want is stored in a variable, so this would become: `_state.locales[transLocale.locale.localeId] = transLocale;`